### ISSUE TITLE
some problems about updating style encoder network

### ISSUE THREAD

**Issue by newExplore-hash** [clovaai/stargan-v2#44]
d_loss, d_losses_latent = compute_d_loss(
    nets, args, x_real, y_org, y_trg, z_trg=z_trg, masks=masks)
self._reset_grad()
d_loss.backward()
optims.discriminator.step()

d_loss, d_losses_ref = compute_d_loss(
    nets, args, x_real, y_org, y_trg, x_ref=x_ref, masks=masks)
self._reset_grad()
d_loss.backward()
optims.discriminator.step()

Hi, I was very curious why the style encoder isn't updated when computing the loss with reference images.

**Comment by toculin38** [comment]
You can't update style encoder when training the discriminator.

**Comment by newExplore-hash** [comment]
> You can't update style encoder when training the discriminator.

no, this is generator, you can find this code in solver.py(line122 - 134 )

**Comment by toculin38** [comment]
> no, this is generator, you can find this code in solver.py(line122 - 134 )

Well, you quoted the discriminator part before you edited it.

To your question, my inference is probably for some consistency issue, style encoder will learn to output similar results for any image as input to minimize the loss "Style Reconstruction" term.

**Comment by yunjey** [comment]
@newExplore-hash Sorry for a late reply. We observed that the style reconstruction loss in latent-guided synthesis is enough to guide the style encoder to extract the styles of reference images. When we trained the style encoder with other losses, we found that the performance was highly degraded.
